controllers/configuration/backend/backend.go

```diff
@@ -54,49 +55,20 @@ type Backend interface {
 }
 
 type backendInitFunc struct {
-	initFuncFromHCL  func(*ParsedBackendConfig, client.Client) (Backend, error)
-	initFuncFromConf func(interface{}, client.Client) (Backend, error)
-}
-
-// ParsedBackendConfig is a struct parsed from the backend hcl block
-type ParsedBackendConfig struct {
-	// Name is the label of the backend hcl block
-	// It means which backend type the configuration will use
-	Name string `hcl:"name,label"`
-	// Attrs are the key-value pairs in the backend hcl block
-	Attrs hcl.Body `hcl:",remain"`
-}
-
-func (conf ParsedBackendConfig) getAttrValue(key string) (*cty.Value, error) {
-	attrs, diags := conf.Attrs.JustAttributes()
-	if diags.HasErrors() {
-		return nil, diags
-	}
-	attr := attrs[key]
-	if attr == nil {
-		return nil, fmt.Errorf("cannot find attr %s", key)
-	}
-	v, diags := attr.Expr.Value(nil)
-	if diags.HasErrors() {
-		return nil, diags
-	}
-	return &v, nil
-}
-
-func (conf ParsedBackendConfig) getAttrString(key string) (string, error) {
-	v, err := conf.getAttrValue(key)
-	if err != nil {
-		return "", err
-	}
-	result := ""
-	err = gocty.FromCtyValue(*v, &result)
-	return result, err
+	initFuncFromHCL  func(ctx k8sContext, backendConfig *ParsedBackendConfig, optionSource *OptionSource) (Backend, error)
+	initFuncFromConf func(ctx k8sContext, backendConfig interface{}, optionSource *OptionSource) (Backend, error)
 }
 
 // ParseConfigurationBackend parses backend Conf from the v1beta2.Configuration
-func ParseConfigurationBackend(configuration *v1beta2.Configuration, k8sClient client.Client) (Backend, error) {
+func ParseConfigurationBackend(configuration *v1beta2.Configuration, k8sClient client.Client, optionSource *OptionSource) (Backend, error) {
 	backend := configuration.Spec.Backend
 
+	ctx := k8sContext{
```

Comment:
In other functions all over this project, ctx is referenced to be an object of context.Context, so it's better not to create another ctx:)

Reply:
OK. 

Actually, I want to use the `k8sContext` to reduce the number of arguments passed into the `newXXXBackend` function. I'll try to find a better way to do this.

Reply:
Parsing ctx, k8sClient, namespace to a function, like others, is okay and consistent.